### PR TITLE
if external storage has been disabled, a full screen error is not shown

### LINKED ISSUE #27285: Do not show a full screen error when 'Allow users to mount external storage' is disabled and we try to configure a mount via Files view
### Steps to reproduce

1.  As admin user, configure a oC server with 'Allow users to mount external storage' disabled (personal mounts)
2. As regular user click in files view in External Storage section
3. Click on 'You can add external storages in the personal settings' link

### Expected behaviour

A warning should be should or the link should not be visible

### Actual behaviour

An error is shown full screen (see gif below)

### Server configuration

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
7.0

**ownCloud version:**

"version":"10.0.0.0"

**Updated from an older ownCloud or fresh install:**
Fresh

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
No

**Logs**

```
```

### Client configuration

**browser**

 Mozilla

errormext

### PR DESCRIPTION
**Pull request by noveens**
## Description

Fixes #27285 

if external storage has been disabled, a full screen error is not shown

## Related Issue

https://github.com/owncloud/core/issues/27285

## Motivation and Context

if external storage has been disabled, a full screen error is not shown

## How Has This Been Tested?

Tested with both admin and non-admin user.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/files_external/templates/list.php [hunk 1]
```diff
@@ -7,7 +8,24 @@
 <div id="emptycontent" class="hidden">
 	<div class="icon-external"></div>
 	<h2><?php p($l->t('No external storage configured')); ?></h2>
-	<p><a href="<?php p(link_to('', 'index.php/settings/admin?sectionid=storage' )); ?>"><?php p($l->t('You can add external storages in the personal settings')); ?></a></p>
+
+	<?php
+		if (OC_User::isAdminUser(OC_User::getUser())) {
```
Comment (PVince81) [inline apps/files_external/templates/list.php]: This is private API. If it contains underscores.

Public API starts with `\OCP\` or `\OC::$server->...`.

Use this instead: `\OC::$server->getGroupManager()->isAdmin($userId)`.
And to get the currently logged in user id: `\OC::$server->getUserSession()->getUser()->getUID()`

Comment (noveens) [inline apps/files_external/templates/list.php]: Didn't know OC_User was also private,

Thanks for the clarification, will update :-)

**Comment by noveens** [comment]
@PVince81 ,
Is that okay now?

File: apps/files_external/templates/list.php [hunk 2]
```diff
@@ -7,7 +7,25 @@
 <div id="emptycontent" class="hidden">
 	<div class="icon-external"></div>
 	<h2><?php p($l->t('No external storage configured')); ?></h2>
-	<p><a href="<?php p(link_to('', 'index.php/settings/admin?sectionid=storage' )); ?>"><?php p($l->t('You can add external storages in the personal settings')); ?></a></p>
+
+	<?php
+		$UserID = \OC::$server->getUserSession()->getUser()->getUID();
```
Comment (PVince81) [inline apps/files_external/templates/list.php]: `$userId`, first letter must be lowercase for variables. Only class names start with a capital letter.

File: apps/files_external/templates/list.php [hunk 3]
```diff
@@ -7,7 +7,25 @@
...
+				'">' .
+				$l->t('You can add external storages in the personal settings') .
+				'</a></p>';
+		}
+
+		else {
+			echo 	'<p><a href="' .
+				link_to('', 'index.php/settings/personal?sectionid=storage' ) .
+				'">' .
+				$l->t('You can add external storages in the personal settings') .
```
Comment (PVince81) [inline apps/files_external/templates/list.php]: I suggest to rename "personal settings" to "settings" so we only have a single message for both user types.

**Comment by PVince81** [comment]
Tested, works.

I realize that the message "External storage is disabled" in the personal page is not correct. An admin can mount external storages and these can be seen by regular users. A more exact formulation for regular users would be "The administrator disabled the ability for regular users to mount external storages". But that's a bit long.

File: apps/files_external/templates/list.php (hunk 3 above)
Comment (noveens) [inline apps/files_external/templates/list.php]: "personal settings" is also there for both user types.
I didn't get you.

Comment (PVince81) [inline apps/files_external/templates/list.php]: yes, but for the admin part you redirect to the admin settings, not personal settings, so it doesn't match the message

Comment (noveens) [inline apps/files_external/templates/list.php]: Oh okay, I got you.

**Comment by noveens** [comment]
@PVince81 ,
commits squashed ;-)
Please have a look.

File: apps/files_external/templates/list.php [hunk 4]
```diff
@@ -7,7 +7,25 @@
...
 	<h2><?php p($l->t('No external storage configured')); ?></h2>
-	<p><a href="<?php p(link_to('', 'index.php/settings/admin?sectionid=storage' )); ?>"><?php p($l->t('You can add external storages in the personal settings')); ?></a></p>
+
+	<?php
+		$userId = \OC::$server->getUserSession()->getUser()->getUID();
+		if (\OC::$server->getGroupManager()->isAdmin($userId)) {
+			echo 	'<p><a href="' .
+				link_to('', 'index.php/settings/admin?sectionid=storage' ) .
+				'">' .
+				$l->t('You can add external storages in storage settings') .
```
Comment (PVince81) [inline apps/files_external/templates/list.php]: "in the storage settings" sounds better ?

**Comment by PVince81** [comment]
:+1:

**Pull request merged**